src/main/java/studio/magemonkey/fusion/queue/CraftingQueue.java

```diff
@@ -1,22 +1,116 @@
 package studio.magemonkey.fusion.queue;
 
 import lombok.Getter;
-import lombok.Setter;
 import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.scheduler.BukkitTask;
+import studio.magemonkey.codex.CodexEngine;
+import studio.magemonkey.codex.api.DelayedCommand;
+import studio.magemonkey.codex.util.messages.MessageData;
+import studio.magemonkey.codex.util.messages.MessageUtil;
+import studio.magemonkey.fusion.*;
+import studio.magemonkey.fusion.cfg.PConfigManager;
+import studio.magemonkey.fusion.cfg.ProfessionsCfg;
+import studio.magemonkey.fusion.util.PlayerUtil;
 
-import java.util.LinkedList;
-import java.util.UUID;
+import java.util.*;
 
+@Getter
 public class CraftingQueue {
 
-    @Getter
-    @Setter
-    private UUID owner;
+    private final Player player;
+    private final String profession;
+    private final Category category;
+    private final List<QueueItem> queue = new ArrayList<>();
+    private final HashMap<Integer, QueueItem> queuedItems;
+    private final BukkitTask queueTask;
 
-    @Getter
-    private final LinkedList<QueueItem> queue = new LinkedList<>();
+    public CraftingQueue(Player player, String profession, Category category) {
+        this.player = player;
+        this.profession = profession;
+        this.category = category;
+        this.queuedItems = new HashMap<>(20);
```

Comment:
Is this Map actually being populated?

Reply:
Yes, its being populated via Lombok Getter in the PlayerCustomGUI three times via `this.queue.getQueuedItems()`.